### DISCUSSION TITLE
How to do first time setup when SWAG is already setup (nginx, lets encrypt etc)

### DISCUSSION THREAD

**Discussion by Glynax** [dani-garcia/vaultwarden#3112] [Q&A] [answered]
So I'm a docker compose user but I'm rather stuck on the examples. Every guide for bit/vaultwarden seem to assume you don't already have a reverse proxy or cert authority and domain -already- set up. I'm sure I can muddle through everything if I could just get past the initial pain point of it being bundled with literally everything. SWAG even has prebuilt sample configs for bit/vaultwarden.
So how is everyone switching to this with an already established system?

**Comment by BlackDex** [comment, marked as answer]
Just start a container with Vaultwarden. Configure your current reverse proxy to point to that container and you are done.

If you already have a reverse proxy just use that, adjust the config or add config. There are some proxy examples on the wiki of this repo if you still need some info.

**Reply by Glynax** [reply]
Sorry it's taken me so long to try that, I have the reverse proxy set up and pointing at it but with the vaultwarden container running I get err too many redirects, tried clearing cookies same error. I do see in the logs that the authorization failed for my hostname, not sure why. Could you link a specific example? I've looked around the wiki but nothing is particularly clicking into place

**Reply by stefan0xC** [reply]
See https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples 
Since you're already using nginx I'd recommend the `Nginx (by blackdex)` example.

**Reply by Glynax** [reply]
Just tried blackdex's config and now I get refused to connect. I think the issue is that caddy just won't authenticate my domain, because I've never been able to access vaultwarden locally either. Haven't seen a first time config or anything I just docker compose up and get stuck

**Reply by stefan0xC** [reply]
I think you need to either [publish the ports](https://stackoverflow.com/q/36320172/6058571) of the service (e.g. to `127.0.0.1:8080:80` and `127.0.0.1:3012:3012`) or replace 127.0.0.1 with the IP of the docker container (which you can find out by running [`docker inspect`](https://stackoverflow.com/a/20686101/6058571)).

**Reply by Glynax** [reply]
So in my compose Caddy is publishing 80:80 and 443:443, I was wondering why the example compose didn't publish 3012 but either way.. Do I even need caddy if I have my own nginx set up? I just tried commenting it out and publishing 80 and 3012 for just vaultwarden, then connecting to the local IP and 80 doesn't work but 3012 atleast gives me this in the logs, with an err connection closed on my browser.
![image](https://user-images.githubusercontent.com/7043861/213792157-d88154d3-d9ca-432c-af29-0d84bf779310.png)

**Reply by Glynax** [reply]
Alright this was all just a big my bad. After going around and around in circles I finally noticed my qbittorrent was using one of the same ports on that box, so of course it wasn't gonna work. Not sure why docker didn't error on launch as it normally does but I'm up and running now. Thanks guys.
